src/net.cpp

```diff
@@ -2539,12 +2540,19 @@ std::vector<CAddress> CConnman::GetAddresses(size_t max_addresses, size_t max_pc
     return addresses;
 }
 
-std::vector<CAddress> CConnman::GetAddresses(Network requestor_network, size_t max_addresses, size_t max_pct)
+std::vector<CAddress> CConnman::GetAddresses(CNode& requestor, size_t max_addresses, size_t max_pct)
 {
+    SOCKET socket;
+    WITH_LOCK(requestor.cs_hSocket, socket = requestor.hSocket);
```

Comment:
81b00f87800f40cb14f2131ff27668bd2bb9e551

Not really sure what is the purpose of `cs_hSocket`, but looks like this should be
```cpp
vector<unsigned char> local_socket_bytes = WITH_LOCK(requestor.cs_hSocket, return GetBindAddress(requestor.hSocket).GetAddrBytes())

Reply:
True, this looks better.
I'd keep it the way it is for now to save the review energy if you don't mind.
I will update these lines if I have other reasons to update the PR.

Reply:
It's fine, can come next too if it's the only thing that needs to be addressed.